src/e2e/step-definitions/external-systems.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Lifely
+ * SPDX-License-Identifier: EUPL-1.2+
+ */
+
+import { Then, When } from "@cucumber/cucumber";
+import { Page } from "@playwright/test";
+import { CustomWorld } from "../support/worlds/world";
+
+const ONE_MINUTE_IN_MS = 60_000;
+
+// Reference to smartdocuments page
+let smartDocumentsPage: Page;
+
+When(
+  "Employee {string} clicks on Create Document for zaak",
+  { timeout: ONE_MINUTE_IN_MS },
+  async function (this: CustomWorld, user) {
+    await this.page.getByText("note_addDocument maken").click();
+
+    const sidebar = this.page.locator("div.sidenav-title");
+    await sidebar.waitFor({ state: "visible" });
+    await sidebar.getByText("Document maken");
+  },
+);
+
+When(
+  "Employee {string} fills in the create document form",
+  { timeout: ONE_MINUTE_IN_MS },
+  async function (this: CustomWorld, user) {
+    const submitButton = this.page.locator("#opslaan_button");
+    await submitButton.waitFor({ state: "visible" });
+
+    // Verify that the button is initially disabled
+    await submitButton.isDisabled();
+
+    // filling the create document form
+    await this.page.getByLabel("Sjabloongroep").click();
+    await this.page
+      .getByRole("option", { name: "Melding evenement organiseren behandelen" })
+      .click();
+    await this.page.waitForTimeout(1000);
+
+    await this.page.getByLabel("Sjabloon").last().click();
+    await this.page.getByRole("option", { name: "OpenZaakTest" }).click();
+
+    const autofillInputTitle = this.page.locator("#title_tekstfield");
+    await autofillInputTitle.click();
+    await autofillInputTitle.fill("Document Title Text");
+
+    // Now check if the button is enabled after those actions
+    await submitButton.isEnabled();
+  },
+);
+
+When(
+  "Employee {string} submits the form to create the document should see SmartDocuments tab",
+  { timeout: ONE_MINUTE_IN_MS },
+  async function (this: CustomWorld, user) {
+    await this.page.click("#opslaan_button");
+
+    smartDocumentsPage = await this.page.waitForEvent("popup");
```

Comment:
I agree with Edgar's comment that this name is not descriptive enough, I'll make a suggestion on the name in another comment.

Reply:
Fixed (in next push)